### DISCUSSION TITLE
Is there a way to provide an optional parameter?

### DISCUSSION THREAD

**Discussion by rezicosta** [vercel/next.js#12367] [Help]
In react I do it this way : `<Route path="/:param(value)?" component={MainComponent} />` Is it achievable in Next? My goal is to check if the param language is "ge", if it is "ge" then pass it in the route, otherwise pass nothing, but still redirect. Final urls should look like this : If "ge" passed - `/ge/news` (Get data in Georgian), if not - `/news` (Get data in English).

**Comment by webNeat** [comment]
I don't know if Next supports optional parameters in the routes, but I think you can achieve what you want in the following way:
First you create a component (not a page) that takes the language code in the props
```jsx
function News({language}) {
    // show news with that language
}
```
Then you create two pages
- `pages/[language]/news.js` in which you get the language from the router params and call the `News` component with it.
- `pages/news.js` in which you simply call `News` and pass `en` as the language.

I hope this helps :)
